modules/e2ee/E2EEncryption.js

```diff
@@ -118,238 +58,19 @@ export class E2EEncryption {
-            }
-            break;
-        }
-    }
-
-    /**
-     * Advances the current key by using ratcheting.
-     *
-     * @private
-     */
-    async _ratchetKeyImpl() {
-        logger.debug('Ratchetting key');
-
-        const material = await importKey(this._key);
-        const newKey = await ratchet(material);
-
-        this._key = new Uint8Array(newKey);
-
-        const index = this._olmAdapter.updateCurrentKey(this._key);
-
-        this._e2eeCtx.setKey(this.conference.myUserId(), this._key, index);
-    }
-
-    /**
-     * Rotates the local key. Rotating the key implies creating a new one, then distributing it
-     * to all participants and once they all received it, start using it.
-     *
-     * @private
-     */
-    async _rotateKeyImpl() {
-        logger.debug('Rotating key');
-
-        this._key = this._generateKey();
-        const index = await this._olmAdapter.updateKey(this._key);
-
-        this._e2eeCtx.setKey(this.conference.myUserId(), this._key, index);
-    }
-
-    /**
-     * Setup E2EE for the receiving side.
-     *
-     * @private
-     */
-    _setupReceiverE2EEForTrack(tpc, track) {
-        if (!this._enabled) {
-            return;
-        }
-
-        const receiver = tpc.findReceiverForTrack(track.track);
-
-        if (receiver) {
-            this._e2eeCtx.handleReceiver(receiver, track.getType(), track.getParticipantId());
-        } else {
-            logger.warn(`Could not handle E2EE for ${track}: receiver not found in: ${tpc}`);
-        }
-    }
-
-    /**
-     * Setup E2EE for the sending side.
-     *
-     * @param {JingleSessionPC} session - the session which sends the media produced by the track.
-     * @param {JitsiLocalTrack} track - the local track for which e2e encoder will be configured.
-     * @private
-     */
-    _setupSenderE2EEForTrack(session, track) {
-        if (!this._enabled) {
-            return;
-        }
-
-        const pc = session.peerconnection;
-        const sender = pc && pc.findSenderForTrack(track.track);
-
-        if (sender) {
-            this._e2eeCtx.handleSender(sender, track.getType(), track.getParticipantId());
-        } else {
-            logger.warn(`Could not handle E2EE for ${track}: sender not found in ${pc}`);
-        }
-    }
-
-    /**
-     * Setup E2EE on the sender that is created for the unmuted local video track.
-     * @param {JitsiLocalTrack} track - the track for which muted status has changed.
-     * @private
-     */
-    _trackMuteChanged(track) {
-        if (browser.doesVideoMuteByStreamRemove() && track.isLocal() && track.isVideoTrack() && !track.isMuted()) {
-            for (const session of this.conference._getMediaSessions()) {
-                this._setupSenderE2EEForTrack(session, track);
-            }
-        }
+     * @param keyInfo: {Object {
+                encryptionKey: CryptoKey
+                index: Number
+        }}
+     * @returns {void}
+    */
+    setEncryptionKey(keyInfo) {
```

Comment:
Checking if setKey is available reads a bit weird. How about this: let's make the base class throw an error that says "not implemented". Olm won't override it. The other one will. This way someone will notice if they are using the API wrong.